sashimi-webapp/src/database/storage.js

```diff
@@ -139,11 +123,13 @@ function creationOfTables() {
 export default class storage {
   static constructor() {}
 
-  static initializeDatabase() {
+  static initializeDatabase(newDatabaseName) {
     if (typeof Promise === 'function') {
       return new Promise((resolve, reject) => {
-        entitiesCreator.initializeDatabase(databaseName)
-        .then(step1 => creationOfTables()
+        databaseName = newDatabaseName || databaseName;
```

Comment:
There might be an issue here if both `databaseName` and `newDatabaseName` are `null` or `undefined`